src/terra-dev-site/guide/DockerDevelopmentEnvironment.i.guide.md

```diff
@@ -0,0 +1,91 @@
+## Docker Development Environment
+
+Terra provides a docker development environment that allows developers access to a common environment to develop Terra components regardless of the OS they are developing on.
+
+In each Terra repository there is a docker-compose.yml file that defines several services, most importantly ```dev``` and ```test```.
+
```

Comment:
Let's start with Terra's docker requirements and call out which docker-compose version is in use. 

> Install Docker version 17.09.0 or higher. Installation instructions can be found at https://docs.docker.com/install/

Then it might be useful to attach some quick resources at the bottom for docker commands: 
- https://docs.docker.com/engine/reference/commandline/cli/
- https://docs.docker.com/compose/reference/overview/

Reply:
It might be helpful to call out some useful commands 
```
docker system prune

// list all existing containers
docker container ls -a

// list current processes
docker ps -a

// use --name to name a docker container
docker run --name my_new_changes <image>

// --rm to immediately remove a container to prevent container bloating
docker run --rm <image>
```

I know a lot of these are called out in the Docker docs....but those doc stink.

Reply:
also `docker images` is useful to see last image build's name

Reply:
I don't really want to explain everything about how to use docker or docker-compose here. Just how to use it in reference to the development container.

Reply:
Or maybe we link to : https://github.com/wsargent/docker-cheat-sheet. Or a good resources for people to get started with learning docker if they don't know

Reply:
I'm fine with linking to a cheat sheet, but I wouldn't go further than that. If we go towards documenting usage of third party tools, then our teams has added not only maintenance, but the responsibility of educating consuming teams as well about third party tools.